service/src/test/java/io/nuvalence/workmanager/service/service/AllowedLinkServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package io.nuvalence.workmanager.service.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.nuvalence.workmanager.service.domain.transaction.AllowedLink;
+import io.nuvalence.workmanager.service.domain.transaction.TransactionLinkType;
+import io.nuvalence.workmanager.service.repository.AllowedLinkRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@ExtendWith(MockitoExtension.class)
+class AllowedLinkServiceTest {
+    @Mock private AllowedLinkRepository repository;
+
+    @Mock private TransactionLinkTypeService transactionLinkTypeService;
+
+    private AllowedLinkService allowedLinkService;
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.openMocks(this);
+        allowedLinkService = new AllowedLinkService(repository, transactionLinkTypeService);
+    }
+
+    @Test
+    void testGetAllowedLinksByDefinitionKey() {
+        String definitionKey = "myDefinitionKey";
+
+        // Mocking the repository
+        List<AllowedLink> expectedLinks = createExpectedLinks();
+        when(repository.findByDefinitionKey(eq(definitionKey))).thenReturn(expectedLinks);
```

Comment:
sonarqube recommendation
```suggestion
        when(repository.findByDefinitionKey(definitionKey)).thenReturn(expectedLinks);
```

Reply:
suggestion applied